src/rest/rest-api-service.js

```diff
@@ -4,14 +4,24 @@ import ApiResponse from './api-response.js';
 class RestApiService {
   constructor(baseUrl) {
     this.baseUrl = baseUrl;
+    this.headers = {
+      'Content-Type': 'application/json',
+    };
   }
 
-  fetch(method, endpoint) {
-    return fetch(`${this.baseUrl}${endpoint}`, { method }).then(response => {
+  resolveUrl(endpoint) {
+    return `${this.baseUrl}${endpoint}`;
+  }
+
+  fetch(method, endpoint, payload = false) {
```

Comment:
Payload is an object, so using false will brake it in typescript. It should be null or undefined by default